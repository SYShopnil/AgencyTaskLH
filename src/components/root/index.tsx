import Button from './button';

export { Button };
export { BannerSectionBar } from './banner-bar';
export { SlideCard } from './slide-card/';
export { HeroSectionWithOnScrollRotation } from './hero-section-with-on-scroll-rotaition';
export { Navbar } from './navbar';
