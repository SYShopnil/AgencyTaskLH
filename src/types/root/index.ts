export type { IButton, INormalButton, IRedirectButton } from './Button/';

export { BtnColorSchema } from './Button/';

export type { IBtnCard } from './btn-card';
export { ColorSchema, CardSize } from './btn-card';

export type { IIconStore } from './_icon';
export { IconName as IconEnums } from './_icon';

export type { ICurveFrame } from './curveFrame';
export { CurveFrameVariantEnum } from './curveFrame';

export type { IInfoCard } from './infoCard';
export type { ITitleDescSection } from './title_desc_sec/';

export type { IBanner, IBannerSectionBar } from './banner-bar/banner.ts';
export type { ISlideCard } from './slide-card';

export type { IHeroSectionWithOnScrollRotation } from './hero-section-with-on-scroll-rotaition/index';
export type { IFooter } from './footer';

export type {
  ICommonSection,
  IMegaMenu,
  IMegaMenuItems,
  ISubMenuBlock,
  INavbar,
} from './navbar';
