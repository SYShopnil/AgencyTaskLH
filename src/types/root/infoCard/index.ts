export interface IInfoCard {
  height: string;
  category: string;
  weight: string;
  abilities: string[];
  dataTestId?: string;
}
