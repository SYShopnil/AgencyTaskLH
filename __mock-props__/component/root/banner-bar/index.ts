import { IBanner, IBannerSectionBar } from '@src/types/root';
import { IconName } from '@src/types/root/_icon';

export const BannerImageMockData: IBanner[] = [
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
];

export const BannerBarSectionMockData: IBannerSectionBar = {
  contents: BannerImageMockData,
  bannerBarSectionTestId: 'mainWrapper',
};
