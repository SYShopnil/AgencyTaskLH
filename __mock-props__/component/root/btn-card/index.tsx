import { CardSize, ColorSchema, IBtnCard } from '@src/types/root';

export const btnCardMockProps: IBtnCard[] = [
  {
    colorSchema: ColorSchema.PrimaryBlueBgTextWhite,
    title: 'Grass',
    cardSize: CardSize.card_lg,
    fill: 65,
    data_test_id: 'btnCardOne',
  },
  {
    colorSchema: ColorSchema.PrimaryBlueBgTextWhite,
    title: 'Grass',
    cardSize: CardSize.card_sm,
    fill: 50,
    data_test_id: 'btnCardTwo',
  },
];
