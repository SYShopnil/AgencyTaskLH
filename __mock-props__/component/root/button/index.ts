import { BtnColorSchema, IRedirectButton } from '@src/types/root';

export const mock_data_redirectButton: IRedirectButton[] = [
  {
    btnText: 'Back to Homepage',
    colorSchema: BtnColorSchema.BlackBgTextWhite,
    isOpenNewTab: true,
    redirectLink: 'https://chat.openai.com/',
    dataTestId: 'Redirect Button One',
  },
  {
    btnText: 'Back to Homepage Two',
    colorSchema: BtnColorSchema.BlackBgTextWhite,
    isOpenNewTab: false,
    redirectLink: '',
    dataTestId: 'Redirect Button Two',
  },
];
