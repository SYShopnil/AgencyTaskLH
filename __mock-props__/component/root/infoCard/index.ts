import { IInfoCard } from '@src/types/root';

export const mock_data_infoCard: IInfoCard = {
  abilities: [`Overgrow`, `Blaze`],
  category: 'Seed',
  height: "2' 04",
  weight: `15.2`,
  dataTestId: 'test-one-for-infoCard',
};
