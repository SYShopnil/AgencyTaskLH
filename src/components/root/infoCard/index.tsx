import React from 'react';
import ModuleStyle from './index.module.scss';
import CurveFrame from '../curveFrame';
import { CurveFrameVariantEnum, IInfoCard } from '@src/types/root';

export const InfoCard = ({
  abilities,
  category,
  height,
  weight,
  dataTestId,
}: IInfoCard) => {
  const publishedData = [
    {
      title: 'Height',
      attr: height || 'N/A',
    },
    {
      title: 'Category',
      attr: category || 'N/A',
    },
    {
      title: 'Weight',
      attr: weight || 'N/A',
    },
    {
      title: 'Abilities',
      abilities,
    },
  ];

  return (
    <span data-testid={dataTestId}>
      <CurveFrame
        borderWidth="0.313rem"
        variant={CurveFrameVariantEnum.with_gradient}
      >
        <div
          className={`${
            ModuleStyle[`container`]
          } py-[2.438rem] md:py-[2.25rem]  pl-[2.877rem]  md:pl-[3rem] bg-[#FFFFFF] duration-[0.5s] transition-all `}
        >
          {publishedData.map((finalDataWhichWillUse, ind) => {
            return (
              <div className={`${ModuleStyle['element']}`} key={ind}>
                <h3 className={`${ModuleStyle['title']}`}>
                  {finalDataWhichWillUse.title}
                </h3>
                {/* print if ability provides */}
                {finalDataWhichWillUse?.abilities && (
                  <>
                    {finalDataWhichWillUse.abilities.map((ability, ind) => {
                      return (
                        <p key={ind} className={`${ModuleStyle['attribute']}`}>
                          {ability}
                        </p>
                      );
                    })}
                  </>
                )}
                {/* print others attributes */}
                {finalDataWhichWillUse?.attr && (
                  <span className={`${ModuleStyle['attribute']} block`}>
                    {finalDataWhichWillUse.attr}
                  </span>
                )}
              </div>
            );
          })}
        </div>
      </CurveFrame>
    </span>
  );
};
