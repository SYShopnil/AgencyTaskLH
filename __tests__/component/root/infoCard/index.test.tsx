import { mock_data_infoCard } from '@root/__mock-props__/component/root/infoCard';
import { InfoCard } from '@src/components/root';
import { getByTestId, render } from '@testing-library/react';

describe('When Info Card has been load', () => {
  it('Expect it will render successfully', () => {
    const { getByTestId } = render(<InfoCard {...mock_data_infoCard} />);
    const catchTheCard = getByTestId(mock_data_infoCard.dataTestId || '');
    expect(catchTheCard).toBeInTheDocument();
    expect(catchTheCard).toMatchSnapshot();
  });
});
