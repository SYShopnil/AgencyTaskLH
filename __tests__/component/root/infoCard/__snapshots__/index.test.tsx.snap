// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Info Card has been load Expect it will render successfully 1`] = `
<span
  data-testid="test-one-for-infoCard"
>
  <div
    class="filter_container "
  >
    <div
      class="box_Outer bg-primaryGradient"
      style="padding: 0.313rem;"
    >
      <div
        class="filter_container"
      >
        <div
          class="box_Inner bg-white"
        >
          <span>
            <div
              class="container py-[2.438rem] md:py-[2.25rem]  pl-[2.877rem]  md:pl-[3rem] bg-[#FFFFFF] duration-[0.5s] transition-all "
            >
              <div
                class="element"
              >
                <h3
                  class="title"
                >
                  Height
                </h3>
                <span
                  class="attribute block"
                >
                  2' 04
                </span>
              </div>
              <div
                class="element"
              >
                <h3
                  class="title"
                >
                  Category
                </h3>
                <span
                  class="attribute block"
                >
                  Seed
                </span>
              </div>
              <div
                class="element"
              >
                <h3
                  class="title"
                >
                  Weight
                </h3>
                <span
                  class="attribute block"
                >
                  15.2
                </span>
              </div>
              <div
                class="element"
              >
                <h3
                  class="title"
                >
                  Abilities
                </h3>
                <p
                  class="attribute"
                >
                  Overgrow
                </p>
                <p
                  class="attribute"
                >
                  Blaze
                </p>
              </div>
            </div>
          </span>
        </div>
      </div>
    </div>
    <svg
      height="0"
      style="visibility: hidden; position: absolute;"
      version="1.1"
      width="0"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <filter
          id="goo"
        >
          <fegaussianblur
            in="SourceGraphic"
            result="blur"
            stdDeviation="3"
          />
          <fecolormatrix
            in="blur"
            mode="matrix"
            result="goo"
            values="1 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 19 -9"
          />
          <fecomposite
            in="SourceGraphic"
            in2="goo"
            operator="atop"
          />
        </filter>
      </defs>
    </svg>
  </div>
</span>
`;
